experimental/DrawingCurves/src/DrawingCurves.jl

```diff
@@ -0,0 +1,191 @@
+   # println("res: ",result)
+   result = [r[1] for r in result]
+   sort!(result)
+   return result
+end
+
+function analyse_singularity(f, projy, xpt, ypt; selected_precision::Int)
+   Rxy = parent(f)
+   y = gens(Rxy)[2]
+   x = gens(Rxy)[1]
+   type = :singularity
+   if(xpt[1] == xpt[2])
+      # Exact case
+      fpy = derivative(f, y)
+      fpx = derivative(f, x)
+      issingular = is_zero(Oscar.evaluate(fpy, [xpt[1], ypt[1]])) && is_zero(Oscar.evaluate(fpx, [xpt[1], ypt[1]]))
+      if !issingular
+         type = :ytangent
+      end
+      pts = _real_roots(projy(Oscar.evaluate(f, [Rxy(xpt[1]), y])); selected_precision)
+      singindex = findall(p->p==ypt[1], pts)
+      @assert length(singindex)==1 "Something went wrong for exact solution"
+      return (pts, singindex[1], type)
+   else
+      xbefore = xpt[1]
+      ptsbefore = _real_roots(projy(Oscar.evaluate(f, [Rxy(xbefore), y])); selected_precision)
+      xafter = xpt[2]
+      ptsafter = _real_roots(projy(Oscar.evaluate(f, [Rxy(xafter), y])); selected_precision)
+
+      # Just guessing some precision, not optimal...
+      yinterval = [ypt[1]-1//selected_precision, ypt[2]+1//selected_precision]
+      result = ptsbefore
+      diff = length(ptsbefore)-length(ptsafter)
+      if diff != 0
+         type = :ytangent
+         if abs(diff) != 2
+            type = :broken
+         end
+         # @assert abs(diff) == 2 "Please choose proper generic coordinate change $diff"
+         if diff < 0
+            result = ptsafter
+         end
+      end
+      
+      # Just verify that everything fits with float
+      singy = Float64(result[findfirst(r->yinterval[1]<=r && r<=yinterval[2], result)])
+      singindices = Int[]
+      outr = typeof(result)()
+      singfound = -1
+      for i in 1:length(result)
+         pt = result[i]
+         if yinterval[1]<=pt && pt<=yinterval[2]
+            if singfound == -1
+               push!(outr, pt)
+               singfound = i
+            end
+            # @assert Float64(pt) == singy "Please increase precision $singy $(Float64(pt))"
+            if Float64(pt) != singy
+               type = :broken
+            end
+            push!(singindices, i)
+         else
+            push!(outr, pt)
+         end
+      end
+
+      if diff != 0
+         ls = length(singindices)
+         if ls != 2
+            type = :broken
+         end
+         # @assert length(singindices) == 2 "Please choose proper generic coordinate change $ls"
+      end
+
+      return (outr, singfound, type)
+   end
+end
+
+
+function msolve_sings(I ; info_level::Int=0, precision::Int=128, interval::Bool=true)
+   # Just to remember how to get more debug info
+   # (sings, rs2) = real_solutions(ideal(f) + ideal(derivative(f, y)); info_level=2, precision=selected_precision, interval=true)
+   (sings, _) = Oscar.real_solutions(I; info_level, precision, interval)
+   sort!(sings, lt=((a,b)->isless(a[1],b[1])))
```

Comment:
```suggestion
   sort!(sings; by=first)
```
This would be slightly more idiomatic

Reply:
Thanks, I didn't know that one.